src/natcap/invest/urban_nature_access.py

```diff
@@ -254,103 +282,133 @@ def execute(args):
         },
-        task_name='2SFCA - Convolve population',
-        target_path_list=[file_registry['convolved_population']],
-        dependent_task_list=[
-            kernel_creation_task,
-            population_alignment_task,
-        ])
-
-    greenspace_nodata = pygeoprocessing.get_raster_info(
-        file_registry['greenspace_area'])['nodata'][0]
-    population_nodata = pygeoprocessing.get_raster_info(
-        file_registry['convolved_population'])['nodata'][0]
-    greenspace_population_ratio_task = graph.add_task(
-        pygeoprocessing.raster_calculator,
-        kwargs={
-            'base_raster_path_band_const_list': [
-                (file_registry['greenspace_area'], 1),
-                (file_registry['convolved_population'], 1),
-                (greenspace_nodata, 'raw'),
-                (population_nodata, 'raw')],
-            'local_op': _greenspace_population_ratio,
-            'target_raster_path': file_registry['greenspace_population_ratio'],
-            'datatype_target': gdal.GDT_Float32,
-            'nodata_target': FLOAT32_NODATA
-        },
-        task_name='2SFCA: Calculate R_j greenspace/population ratio',
-        target_path_list=[file_registry['greenspace_population_ratio']],
-        dependent_task_list=[
-            convolved_population_task,
-            greenspace_reclassification_task,
-        ])
-
-    convolved_greenspace_population_ratio_task = graph.add_task(
-        pygeoprocessing.convolve_2d,
+        target_path_list=[file_registry['attribute_table']],
+        priority=10,  # higher-than-normal priority to address sooner.
+        task_name='Preprocess the attribute table')
+
+    preprocess_attr_table_task.join()
+    attr_table = pandas.read_csv(file_registry['attribute_table'])
+    partial_greenspace_paths = []
+    partial_greenspace_tasks = []
+    for search_radius_m, group in attr_table[
+            attr_table['greenspace'] == 1].groupby('search_radius_m'):
+        matching_landuse_codes = group['lucode'].unique()
+        LOGGER.info(
+            f'Using search radius {search_radius_m} for lucodes '
+            f'{" ".join([str(c) for c in matching_landuse_codes])}')
+
+        # reclassify greenspace needed for this kernel
+        greenspace_pixels_path = os.path.join(
+            intermediate_dir, f'greenspace_{search_radius_m}{suffix}.tif')
+        greenspace_reclassification_task = graph.add_task(
+            _reclassify_greenspace_area,
+            kwargs={
+                'lulc_raster_path': file_registry['aligned_lulc'],
+                'lulc_attribute_table': file_registry['attribute_table'],
+                'target_raster_path': greenspace_pixels_path,
+                'only_these_greenspace_codes': set(matching_landuse_codes),
+            },
+            target_path_list=[greenspace_pixels_path],
+            task_name=(
+                f'Identify area of greenspace in pixels {search_radius_m}'),
```

Comment:
I'm not sure what this is saying, maybe missing a word?

Reply:
Ha, this really isn't readable at all, is it?  Greatly simplified in  cfa5bbdd9